### PR TITLE
feat(logging): Add interactionId to header of Cody Client requests (CODY-4117)

### PR DESCRIPTION
**Pull request by julialeex**
This PR does [add interactionId to header of Cody Client requests](https://linear.app/sourcegraph/issue/CODY-4117/add-interactionid-to-header-of-cody-client-requests). 

`interactionId` is already in telemetry, adding this to `cody_gateway` as well.

## Test plan
Added a test in `ChatController.test.ts`.

### CODE REVIEW

File: vscode/src/chat/agentic/DeepCody.test.ts [hunk 1]
```diff
@@ -149,7 +149,7 @@ describe('DeepCody', () => {
             mockCurrentContext
         )
 
-        const result = await agent.getContext({ aborted: false } as AbortSignal)
+        const result = await agent.getContext('', { aborted: false } as AbortSignal)
```
Comment (abeatrix) [inline vscode/src/chat/agentic/DeepCody.test.ts:152]:
```suggestion
        const result = await agent.getContext('deep-cody-test-interaction-id', { aborted: false } as AbortSignal)
```

nit

**Review by abeatrix — APPROVED** [review]
You did it! With these changes do we still need to update the recordings?

File: lib/shared/src/sourcegraph-api/completions/client.ts [hunk 2]
```diff
@@ -87,15 +88,23 @@ export abstract class SourcegraphCompletionsClient {
...
-        const { apiVersion } = requestParams
+    ): Promise<{
+        url: URL
+        serializedParams: SerializedCompletionParameters
+        headerParams: Record<string, string>
+    }> {
+        const { apiVersion, interactionId } = requestParams
         const serializedParams = await getSerializedParams(params)
+        const headerParams: Record<string, string> = {}
+        if (interactionId) {
```
Comment (abeatrix) [inline lib/shared/src/sourcegraph-api/completions/client.ts:99]: It's be helpful to add a comment here about why we want to add the interaction Id here

Comment (julialeex) [reply lib/shared/src/sourcegraph-api/completions/client.ts:99]: ah good point!

**Review by abeatrix — APPROVED** [review]
Saw there was a lint issue so I just run `pnpm run biome` to fix it. It should be good to go now, thanks!

**Comment by julialeex** [comment]
> You did it! With these changes do we still need to update the recordings?

yes

**Pull request merged**